_posts/2020-01-16-spring-bean-life-cycle.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "The Lifecycle of Spring Beans"
+categories: [spring-boot]
+modified: 2020-01-16
+excerpt: "An in-depth look at the paging support provided by Spring Data for querying
+          Spring Web MVC controllers and Spring Data repositories."
+image:
+  auto: 0012-pages
+
+---
+
+## What are Spring Beans?
+ > Spring Beans are Java Objects or Instances which is created and managed by Spring container.
+
+Spring `@ApplicationContext` is responsible for creating a Spring bean and manage the 
+Spring bean life cycle.
+
+Below, We have created a simple Spring bean for `Foo` class using `@Component` annotation.
+
+```java
+@Component
+Class Foo { //variables and methods ...}
+```
+## The Spring Bean Lifecycle
+
+The Bean's life cycle consists of 2 phases `Post-initialization` and `Pre-destruction`. 
+
+Below are the Phase 1, `Post-initialization` steps while creating a Bean.

```

Comment:
The description of each step below is too shallow. Please make them complete sentences explaining each step in more detail. 

You may want to split the post-init and the pre-destroy lifecycle steps each into their own section. Then, in that section, you can use a little more text to describe each step.